docs/src/man/basics.md

```diff
@@ -0,0 +1,2002 @@
+DataFrameRow
+ Row │ Age    Sex     Job
+     │ Int64  String  Int64
+─────┼──────────────────────
+   2 │    85  female      3
+
+julia> dfr
+DataFrameRow
+ Row │ Age    Sex     Job
+     │ Int64  String  Int64
+─────┼──────────────────────
+   2 │    85  female      3
+
+julia> dfr.Age = 98 # set value of col `:Age` in row `2` to `98` in-place
+98
+
+julia> dfr
+DataFrameRow
+ Row │ Age    Sex     Job
+     │ Int64  String  Int64
+─────┼──────────────────────
+   2 │    98  female      3
+
+julia> dfr[2:3] = ["male", 2] # set values of entries in columns `:Sex` and `:Job` 
+2-element Vector{Any}:
+  "male"
+ 2
+
+julia> dfr
+DataFrameRow
+ Row │ Age    Sex     Job
+     │ Int64  String  Int64
+─────┼──────────────────────
+   2 │    98  male        2
+```
+
+As mentioned in the title that we will discuss `setindex!` for *DataFrame*, *SubDataFrame*, *DataFrameRow*. So here
+we are talking about *SubDataFrame* and to create a *SubDataFrame* we need of `view` function. To know more about 
+`view` please have a look to **Views** section of this tutorial. 
+
+```jldoctest dataframe
+julia> sdf = view(df1, :, 2:3) # Column subsetting
+6×2 SubDataFrame
+ Row │ Sex          Job
+     │ String       Int64
+─────┼────────────────────
+   1 │ male             2
+   2 │ male             2
+   3 │ male             4
+   4 │ transgender      2
+   5 │ female           2
+   6 │ male             1
+
+julia> sdf[2, :Sex] = "female" # set value of col `:Sex` in second row to `female` in-place
+"female"
+
+julia> sdf
+6×2 SubDataFrame
+ Row │ Sex          Job
+     │ String       Int64
+─────┼────────────────────
+   1 │ male             2
+   2 │ female           2
+   3 │ male             4
+   4 │ transgender      2
+   5 │ female           2
+   6 │ male             1
+
+julia> sdf[6, 1:2] = ["female", 3] # set value of multiple columns
+2-element Vector{Any}:
+  "female"
+ 3
+
+julia> sdf
+6×2 SubDataFrame
+ Row │ Sex          Job
+     │ String       Int64
+─────┼────────────────────
+   1 │ male             2
+   2 │ female           2
+   3 │ male             4
+   4 │ transgender      2
+   5 │ female           2
+   6 │ female           3
+```
+
+In the above examples we have talked about *SubDataFrame*. Now in the given example we set the value 
+of 3rd-column which is `:Job` in our data frame `df1` with `[4, 5, 7, 8, 2, 1]`:
+
+```jldoctest dataframe
+julia> df1[:, 3] = [4, 5, 7, 8, 2, 1]
+6-element Vector{Int64}:
+ 4
+ 5
+ 7
+ 8
+ 2
+ 1
+
+julia> df1
+6×3 DataFrame
+ Row │ Age    Sex          Job
+     │ Int64  String       Int64
+─────┼───────────────────────────
+   1 │    80  male             4
+   2 │    98  female           5
+   3 │    78  male             7
+   4 │    95  transgender      8
+   5 │    78  female           2
+   6 │    89  female           1
+```
+
+Column `:Age` is replaced freshly allocated vector because of broadcasting assignment:
```

Comment:
If you want to introduce broadcasting assignment here I would first say that apart from normal assignment one can do broadcasting assignment and explain the difference in general.